### PR TITLE
Improve test coverage for OutputCaptureRule

### PR DESCRIPTION
**Pull request by currenjin**
The test was added because there were not enough tests for the method of class OutputCaptureRule.

Method list
- getAll
- getOut
- getErr

### CODE REVIEW

File: spring-boot-project/spring-boot-test/src/test/java/org/springframework/boot/test/system/OutputCaptureRuleTests.java [hunk 1]
```diff
@@ -37,10 +37,27 @@ public void toStringShouldReturnAllCapturedOutput() {
...
 	}
 
+	@Test
+	public void getAllShouldReturnAllCapturedOutput() {
+		System.out.println("Hello World");
+		assertThat(this.output.getAll()).contains("Hello World");
+	}
+
+	@Test
+	public void getOutShouldReturnAllCapturedOutput() {
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-test/src/test/java/org/springframework/boot/test/system/OutputCaptureRuleTests.java:47]: I think this should named something like `getOutShouldOnlyReturnOutputCapturedFromSystemOut`

File: spring-boot-project/spring-boot-test/src/test/java/org/springframework/boot/test/system/OutputCaptureRuleTests.java [hunk 2]
```diff
@@ -37,10 +37,27 @@ public void toStringShouldReturnAllCapturedOutput() {
...
+	}
+
 	@Test
 	public void captureShouldBeAssertable() {
 		System.out.println("Hello World");
 		assertThat(this.output).contains("Hello World");
 	}
 
+	@Test
+	public void getErrShouldReturnAllCapturedOutput() {
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-test/src/test/java/org/springframework/boot/test/system/OutputCaptureRuleTests.java:59]: I think this should named something like `getErrShouldOnlyReturnOutputCapturedFromSystemErr`

File: spring-boot-project/spring-boot-test/src/test/java/org/springframework/boot/test/system/OutputCaptureRuleTests.java [hunk 3]
```diff
@@ -37,10 +37,27 @@ public void toStringShouldReturnAllCapturedOutput() {
 		assertThat(this.output.toString()).contains("Hello World");
 	}
 
+	@Test
+	public void getAllShouldReturnAllCapturedOutput() {
+		System.out.println("Hello World");
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-test/src/test/java/org/springframework/boot/test/system/OutputCaptureRuleTests.java:42]: This test should generate output using both `System.err` and `System.out` and then check that `getAll()` contains all of it.

File: spring-boot-project/spring-boot-test/src/test/java/org/springframework/boot/test/system/OutputCaptureRuleTests.java [hunk 4]
```diff
@@ -37,10 +37,27 @@ public void toStringShouldReturnAllCapturedOutput() {
...
+		assertThat(this.output.getAll()).contains("Hello World");
+	}
+
+	@Test
+	public void getOutShouldReturnAllCapturedOutput() {
+		System.out.println("Hello World");
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-test/src/test/java/org/springframework/boot/test/system/OutputCaptureRuleTests.java:48]: This test should generate output using both `System.err` and `System.out` and then check that `getOut()` only contains the output from `System.out`.

File: spring-boot-project/spring-boot-test/src/test/java/org/springframework/boot/test/system/OutputCaptureRuleTests.java [hunk 5]
```diff
@@ -37,10 +37,27 @@ public void toStringShouldReturnAllCapturedOutput() {
...
+
 	@Test
 	public void captureShouldBeAssertable() {
 		System.out.println("Hello World");
 		assertThat(this.output).contains("Hello World");
 	}
 
+	@Test
+	public void getErrShouldReturnAllCapturedOutput() {
+		System.err.println("Hello Error");
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-test/src/test/java/org/springframework/boot/test/system/OutputCaptureRuleTests.java:60]: This test should generate output using both `System.err` and `System.out` and then check that `getErr()` only contains the output from `System.err`.

**Review by wilkinsona — CHANGES_REQUESTED** [review]
Thanks for the PR, @currenjin. I have left a few review comments for your consideration.

**Comment by currenjin** [comment]
@wilkinsona Thanks for your comment.
It was a good review :D

I applied it, is it okay?

**Comment by philwebb** [comment]
Thanks for the updates @currenjin. I've merged this to `2.7` and `main`.

**Pull request closed without merging**